test_programs/execution_success/unary_operator_overloading/src/main.nr

```diff
@@ -0,0 +1,36 @@
+use std::ops::{Neg, Not};
+
+// x = 3
+fn main(x: u32) {
+    let wx = Wrapper::new(x as i32);
+    let ex: i32 = 3;
+
+    assert((-wx).inner == -ex);
+    assert((!wx).inner == !ex);
+
+    // Check that it works with type variables (x's type isn't immediately known)
+    let x = 3;
+    assert(-3 == -x);
```

Comment:
May want to check:
- Without `x` since the compiler can use that to know `x: i32`
- Without `x` on a `!` operation to ensure the compiler will default to `u32` like it does with other bitwise ops:
```rs
assert_eq(!0, large number);
```